doc/dash-acl-tags/dash-acl-tags.md

```diff
@@ -0,0 +1,277 @@
+```
+DASH_PREFIX_TAG_TABLE:{{tag_name}}
+    "ip_version": {{ipv4/ipv6}}
+    "prefix_list": {{list of prefix}}
+```
+
+```
+tag_name                  = STRING; unique tag name
+ip_version                = IP version (IPv4/IPv6)
+prefix_list               = list of ip prefixes ',' separated. It is valid to have empty list of prefixes.
+                            If the prefix is empty, no packets will be assigned to this TAG.
+```
+
+### DASH_ACL_RULE_TABLE
+The existing `DASH_ACL_RULE_TABLE` will be extended with a new `src_tag` and `dst_tag` optional attributes.
+```
+DASH_ACL_RULE_TABLE:{{group_id}}:{{rule_num}}
+    "priority": {{priority}}
+    "action": {{action}}
+    "terminating": {{bool}}
+    "protocol": {{list of protocols}} (OPTIONAL)
+    "src_tag": {{list of tag name}} (OPTIONAL)
+    "dst_tag": {{list of tag name}} (OPTIONAL)
+    "src_addr": {{list of prefix}} (OPTIONAL)
+    "dst_addr": {{list of prefix}} (OPTIONAL)
+    "src_port": {{list of range of ports}} (OPTIONAL)
+    "dst_port": {{list of range of ports}} (OPTIONAL)
+```
+```
+key                      = DASH_ACL_RULE_TABLE:group_id:rule_num ; unique rule num within the group.
+; field                  = value
+priority                 = INT32 value  ; priority of the rule, lower the value, higher the priority
+action                   = allow/deny
+terminating              = true/false   ; if true, stop processing further rules
+protocols                = list of INT ',' separated; E.g. 6-tcp, 17-udp; if not provided, match on all protocols
+src_tag                  = list of source tag name ',' separated; if not provided, match on ANY tag or NO tag.
+dst_tag                  = list of destination tag name ',' separated; if not provided, match on ANY tag or NO tag.
+src_addr                 = list of source ip prefixes ',' separated; if not provided, match on all source IPs.
+dst_addr                 = list of destination ip prefixes ',' separated; if not provided, match on all destination IPs.
+src_port                 = list of range of source ports ',' separated;  if not provided, match on all source ports.
+dst_port                 = list of range of destination ports ',' separated;  if not provided, match on all destination ports.
+```
+
+## SWSS
+
+The following components of orchagent shall be modified.
+
+## DashAclOrch
+
+### Requirements
+
+- Provide support for adding new ACL tags with the list of prefixes in the DASH_PREFIX_TAG_TABLE APP DB table.
+- Provide support for updating list of prefixes for ACL tags. 
+- Provide support for removing ACL tags. ACL tag can be removed only if it is not attached to any ACL rules.
+- Provide support for multiple source and destination ACL tags for ACL rules in DASH_ACL_RULE_TABLE.
+- Provide support for support for TAGs and list prefixes configuration for ACL rules in DASH_ACL_RULE_TABLE.
+
+### Configuration flow
+
+![Configuration flow](images/dash-acl-tags.png)
```

Comment:
Isn't ACL configured from config_db? I see DASH_ACL_RULE and DASH_ACL_GROUP in config_db schema

Reply:
The schema is for the APP DB. Microsoft had a plan to add configuration validation for APP DB as well.
Right now the schema is outdated. I checked with Prince, and currently, we don't need to update it.